divi/src/merkletx.cpp

```diff
@@ -99,15 +97,20 @@ int CMerkleTx::GetNumberOfBlockConfirmationsINTERNAL(const CBlockIndex*& pindexR
 {
     if (hashBlock == 0 || nIndex == -1)
         return 0;
-    AssertLockHeld(cs_main);
 
     // Find the block it claims to be in
-    BlockMap::const_iterator mi = blockIndices_.find(hashBlock);
-    if (mi == blockIndices_.end())
-        return 0;
-    CBlockIndex* pindex = (*mi).second;
-    if (!pindex || !activeChain_.Contains(pindex))
-        return 0;
+    unsigned bestHeight;
+    CBlockIndex* pindex;
+    {
+        LOCK(cs_main);
+        BlockMap::const_iterator mi = blockIndices_.find(hashBlock);
+        if (mi == blockIndices_.end())
+            return 0;
+        pindex = (*mi).second;
+        if (!pindex || !activeChain_.Contains(pindex))
+            return 0;
+        bestHeight = activeChain_.Height();
```

Comment:
bestHeight isn't used in the return value

Reply:
Good catch, fixed.